lobe_android/app/src/main/java/com/example/test/tflite/Classifier.java

```diff
@@ -159,28 +168,38 @@ protected Classifier(Activity activity, Device device, int numThreads) throws IO
                 }
             } catch (JSONException e) {
             }
+
+            // Reads type and shape of input and output tensors, respectively.
+            JSONObject inputImage = jsonObject.getJSONObject("inputs").getJSONObject("Image");
+            JSONArray shapeArray = inputImage.getJSONArray("shape"); // {1, height, width, 3}
+            String typeValue = inputImage.getString("dtype").toUpperCase();
+            tempImageSizeY = shapeArray.getInt(1);
+            tempImageSizeX = shapeArray.getInt(2);
+            imageDataType = DataType.valueOf(typeValue);
+
+            JSONObject outputConfidences = jsonObject.getJSONObject("outputs").getJSONObject("Confidences");
+            JSONArray outputShapeArray = outputConfidences.getJSONArray("shape"); // {1, NUM_CLASSES}
+            String outputTypeValue = outputConfidences.getString("dtype").toUpperCase();
+            probabilityShape = new int[outputShapeArray.length()];
+            for (int i = 0, l = outputShapeArray.length(); i < l; i++) {
+                try {
+                    probabilityShape[i] = outputShapeArray.getInt(i);
+                } catch (JSONException e) {
+                    probabilityShape[i] = 1; // default
+                }
+            }
+            probabilityDataType = DataType.valueOf(outputTypeValue);
         } catch (JSONException e) {
             e.printStackTrace();
         }
 
-
-        // Reads type and shape of input and output tensors, respectively.
-        int imageTensorIndex = 0;
-        int[] imageShape = tflite.getInputTensor(imageTensorIndex).shape(); // {1, height, width, 3}
-        imageSizeY = imageShape[1];
-        imageSizeX = imageShape[2];
-        DataType imageDataType = tflite.getInputTensor(imageTensorIndex).dataType();
-        // First output is the predicted label
-        // Second output is the array of confidences.
-        int probabilityTensorIndex = 0;
-        int[] probabilityShape =
-                tflite.getOutputTensor(probabilityTensorIndex).shape(); // {1, NUM_CLASSES}
-        DataType probabilityDataType = tflite.getOutputTensor(probabilityTensorIndex).dataType();
+        imageSizeY = tempImageSizeY;
```

Comment:
if this fails, sizes are set to 0,0. would be better to either hard crash if there is an issue, or to make the temp sizes 224, 224

Reply:
Yeah great point. I will go with 224 for now.